Show calendar color in trash bin

st3iny opened the pull request:
Fixes #3175 

~~Requires my over engineered code from https://github.com/nextcloud/cdav-library/pull/492.~~ (merged)

![Screenshot 2021-06-20 at 15-40-28 June 2021 - Calendar - Nextcloud](https://user-images.githubusercontent.com/1479486/122676302-dcf5b580-d1dd-11eb-9379-4cdffdd64326.png)

**EDIT:** Updated screenshot again

jancborchardt commented on src/components/AppNavigation/CalendarList/Trashbin.vue:
```diff
@@ -226,4 +233,13 @@ th {
+	vertical-align: middle;
+	width: 14px;
+	height: 14px;
+	margin: 0 14px;
```
Tiny detail, would look better with only `margin-right: 14px;` so the color dots are aligned with the text above.

Otherwise super nice! :)

st3iny replied:
Thanks for the feedback. I removed the left margin and updated the screenshot.

st3iny commented:
I fixed all conflicts and updated the screenshot. Tests fail until https://github.com/nextcloud/calendar/issues/3221 is released (or rejected). The branch of cdav-library has to be checked out manually until then. Run `npm install /path/to/local/cdav/library` prior to testing.

ChristophWurst commented on src/services/caldavService.js:
```diff
@@ -23,6 +23,7 @@ import DavClient from 'cdav-library'
 import { generateRemoteUrl } from '@nextcloud/router'
 import { getRequestToken } from '@nextcloud/auth'
 import { CALDAV_BIRTHDAY_CALENDAR } from '../models/consts.js'
+import { DeletedCalendar } from 'cdav-library/src/models/deletedCalendar'
```
unused

st3iny replied:
Gone

ChristophWurst commented:
Update https://github.com/nextcloud/cdav-library/pull/492 and then I'll give this another review and test :)

tcitworld commented on src/components/AppNavigation/CalendarList/Trashbin.vue:
```diff
@@ -143,6 +154,7 @@ export default {
 					subline,
 					url: vobject.uri,
 					deletedAt: vobject.dav._props['{http://nextcloud.com/ns}deleted-at'],
+					color: vobject.calendar.color,
```
No fallback to calendar color?

st3iny replied:
We don't have a fallback color in the calendar list either. Should I add one?

tcitworld replied:
I'd say yes, with https://github.com/nextcloud/calendar/blob/master/src/components/AppNavigation/CalendarList/CalendarListNew.vue#L194 for coherence

st3iny replied:
Thanks for the link. I'll add it.

st3iny commented:
> Update [nextcloud/cdav-library#492](https://github.com/nextcloud/cdav-library/pull/492) and then I'll give this another review and test :)

Dependabot still has not opened a PR to update cdav-library so I updated it here manually.

tcitworld commented on src/components/AppNavigation/CalendarList/Trashbin.vue:
```diff
@@ -143,6 +155,7 @@ export default {
 					subline,
 					url: vobject.uri,
 					deletedAt: vobject.dav._props['{http://nextcloud.com/ns}deleted-at'],
+					color: vobject.calendar.color ?? uidToHexColor(vobject.calendar.displayName),
```
I must have missed this.

It should be `vobject.color` (the event's own color), then fallback to the `vobject.calendar.color`, and finally fallback to the color generated from the calendar name.

st3iny replied:
Ahh right, events can have colors too. I thought we always show the color of the calendar :man_facepalming: 

Many thanks for your thorough review :)

tcitworld commented on src/components/AppNavigation/CalendarList/Trashbin.vue:
```diff
@@ -127,8 +139,10 @@ export default {
 					// ignore
 				}
 				let subline = vobject.calendar.displayName
+				let color = null
```
Not sure if needed, `VTODO` and `VJOURNAL` can have colors as well.

st3iny replied:
Done. Now the color of the first component is tried to be used regardless of its type.
